src/hooks/tbtc/useFetchDepositDetails.ts

```diff
@@ -100,5 +117,23 @@ export const useFetchDepositDetails = (depositKey: string | undefined) => {
     }
   }, [depositKey, threshold, reverseTxHash])
 
+  useEffect(() => {
+    setDepositData(
+      (prevState) =>
+        ({
+          ...prevState,
+          optimisticMintingRequestedTxHash,
+          optimisticMintingFinalizedTxHash,
+        } as DepositData)
+    )
+  }, [optimisticMintingFinalizedTxHash, optimisticMintingRequestedTxHash])
+
+  useEffect(() => {
+    setDepositData((prevState) => ({
+      ...prevState,
+      confirmations: txConfirmations,
+    }))
+  }, [txConfirmations])
```

Comment:
I would prefer to not put it here and update like that, because it might lead to unexpected behaviour. For example we can have 5/6 confirmations for the given deposit. Let's say we go to the deposit details page and the fetch starts. We are at this line:
```
const confirmations =
          (await threshold.tbtc.getTransactionConfirmations(btcTxHash)) ??
          txConfirmations
```
and `confirmations` is equal to 5. After that (but before fetch is over) the 6th confirmation happens so we update the `confirmations` in the state with the value of 6. Then, at the end of the fetch, we overwrite the value with `5` which is out-dated by now.

Same can happen with tx hashes.

The safer way would be to move those two calls:
```
const { optimisticMintingFinalizedTxHash, optimisticMintingRequestedTxHash } =
    useSubscribeToOptimisticMintingEvents(depositKey)
  const { txConfirmations } = useTbtcState()
```

to `DepositDetails.tsx` and the use double question mark to choose the right value. The values from the state should have a priority.

Reply:
As long as the value is updated, the change is captured by `useEffect` hook and the `depositData` state is updated accordingly. Notice I'm aggregating the update upon the most recent state value. 
Actually the `isFetching` flag is there only to provide feedback for loading state, nothing more.
Spreading correlated state update across multiple files/hooks is confusing, difficult to maintain and should be avoided. That's why I'm trying to keep everything in one place. If you call `useFetchDepositData` you expect to get **complete** deposit data. How could I know to get confirmations count from another place and transaction hashes from one another place? It's overcomplicated - especially if the shape of data provided by the `useFetchDepositData` hook indicates both confirmations count and hashes are there and (intuitively) should be there up to date but isn't.

If you disagree and you have idea how to resolve it, please feel free to commit your changes. I ain't got a what to do with it 😆

Reply:
I've improvde it by moving the deposit details data to redux store. Please see those commits:

397810d59511727dd61b3d1aaf2f29304052feea (see this one's commit message for more info)
e0df1c7d691a1dbfc8813e6e1b9b6ad269ca449c
94fc793c5b63f51b5011c8e3636967e84da2c719